File: benchmarks/comparison/matrix_mul.rs
fn main() {
    let n: i64 = 1500;
    let mut total: i64 = 0;
    for i in 0..n {
        for j in 0..n {
            let mut sum: i64 = 0;
            for k in 0..n {
                sum = sum.wrapping_add((i * n + k).wrapping_mul(k * n + j).wrapping_add(total ^ k));
            }
            total = total.wrapping_add(sum);
        }
    }
    println!("{}", total);
}

File: benchmarks/comparison/matrix_mul.c
#include <stdio.h>
#include <stdint.h>

int main(void) {
    int n = 1500;
    int64_t total = 0;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++) {
            int64_t sum = 0;
            for (int k = 0; k < n; k++)
                sum += (int64_t)(i * n + k) * (k * n + j) + (total ^ k);
            total += sum;
        }
    printf("%ld\n", total);
    return 0;
}
